include/xtensor/xview.hpp

```diff
@@ -1661,6 +1667,75 @@ namespace xt
         return detail::make_view_impl(std::forward<E>(e), std::make_index_sequence<sizeof...(S)>(), std::forward<S>(slices)...);
     }
 
+    namespace detail
+    {
+        class RowImpl
+        {
+        public:
+            template<class E>
+            static inline auto make(E&& e, const int index)
+            {
+                check_dimension(e.shape());
+                return view(e, index, xt::all());
+            }
+
+        private:
+            template<class S>
+            static inline void check_dimension(const S& shape)
+            {
+                if (shape.size() != 2)
+                {
+                    XTENSOR_THROW(std::invalid_argument, "A row can only be accessed on an expression with exact two dimensions");
+                }
+            }
+
+            template<class T, int N>
+            static inline void check_dimension(const std::array<T, N>&)
+            {
```

Comment:
I don't think you need to make a specialization of `check_dimension` for `std::array`: since it provides a `size` method, the generic implementation should be enough.

Reply:
I added a specialization  for `std::array` to forbid a call on  containers with compile-constant dimensions.

Reply:
Ah indeed, it makes sense to have the error at compile time.